k8srm-prototype/pkg/api/claim_types.go

```diff
@@ -53,66 +37,30 @@ type DeviceClassSpec struct {
 	// +required
 	DeviceType string `json:deviceType,omitempty`
 
-	// Driver specifies the driver that should handle this class of devices.
-	// When a DeviceClaim uses this class, only devices published by the
-	// specified driver will be considered.
-	// +optional
-	Driver *string `json:driver,omitempty`
-
 	// Constraints is a CEL expression that operates on device attributes,
 	// and must evaluate to true for a device to be considered. It will be
 	// ANDed with any Constraints field in the DeviceClaim using this class.
+	//
 	// +optional
 	Constraints *string `json:"constraints,omitempty"`
 
-	// Device claims may represent be satisfied by choosing multiple
-	// devices instead of just a single device. The min/max fields control
-	// whether we want a single device, or a set of devices to satisfy a
-	// claim.
-
-	// MinDeviceCount is the minimum number of devices that should be selected
-	// when satsifying a claim using this class. Default is 1.
-	// +optional
-	MinDeviceCount *int `json:"minDeviceCount,omitempty"`
-
-	// MaxDeviceCount is the maximum number of devices that should be selected
-	// when sastisfying a claim using this class. No maximum, by default.
-	// +optional
-	MaxDeviceCount *int `json:"maxDeviceCount,omitempty"`
-
-	// MatchAttributes allows specifying a constraint within a set of
-	// chosen devices, without having to explicitly specify the value of
-	// the constraint.  For example, this allows constraints like "all
-	// devices must be the same model", without having to specify the exact
-	// model. We may be able to use this for some basic topology
-	// constraints too, by representing the topology as device attributes.
-	//
-	// Currently, these are just strings. However, we could make them
-	// structs, and include required vs preferred matches. Required matches
-	// would fail if not met, where as preferred would lower the score if
-	// not met. We could even allow low/medium/high priority and adjust the
-	// score differently for each.
+	// AdminAccess indicates that this class provides administrative access
+	// to the devices. Claims using a class with AdminAccess are expected
+	// to be used for monitoring or other management services for a device.
+	// They ignore all ordinary claims to the device with respect to access
+	// modes and any resource allocations. Access to these classes must be
+	// controlled via ResourceQuota. Default is false.
 	//
 	// +optional
-	MatchAttributes []string `json:"matchAttributes,omitempty"`
+	AdminAccess *bool `json:"adminAccess,omitempty"`
```

Comment:
to use in quota, probably has to move to the claim

Reply:
I'm still fuzzy on how to model this, and how to indicate when it doesn't apply (e.g. a netdevice).  That said, we NEED to pin down access control/quota or this whole thing is dubious.